pkg/minikube/cluster/cluster.go

```diff
@@ -181,27 +181,124 @@ func StartCluster(h sshAble) error {
 	return nil
 }
 
-type fileToCopy struct {
+type CopyableFile interface {
+	io.Reader
+	GetLength() int
+	GetAssetName() string
+	GetTargetDir() string
+	GetTargetName() string
+	GetPermissions() string
+}
+
+type FileAsset struct {
+	Data        []byte
+	Reader      io.Reader
+	Length      int
 	AssetName   string
 	TargetDir   string
 	TargetName  string
 	Permissions string
 }
 
-var assets = []fileToCopy{
-	{
+func (f *FileAsset) Read(p []byte) (int, error) {
+	if f.Reader == nil {
+		contents, err := ioutil.ReadFile(f.AssetName)
+		if err != nil {
+			return 0, err
+		}
+		f.Data = contents
+		f.Length = len(contents)
+		f.Reader = bytes.NewReader(f.Data)
+	}
+	return f.Reader.Read(p)
+}
+
+// unsure if this should return special case int values or int, error
+func (f *FileAsset) GetLength() int {
```

Comment:
I think you could make a "BaseAsset" struct that has these methods, and "embed" it like you did with io.Reader. Then you wouldn't need to redefine the boilerplate ones around AssetName and TargetDir twice.